src/server/watchServerChanges.js

```diff
@@ -8,82 +8,87 @@ import generateDefaultConfig from './generateDefaultConfig';
+    //       if (err.details) {
+    //         console.error(err.details);
+    //       }
+    //       return;
+    //     }
+    //     const info = stats.toJson();
+    //     if (stats.hasErrors()) {
+    //       console.error(info.errors);
+    //     }
+    //
+    //     if (stats.hasWarnings()) {
+    //       console.warn(info.warnings);
+    //     }
+    //   }
+    // });
 
-  const compilerOptions = {
-    aggregateTimeout: 300, // wait so long for more changes
-    poll: true, // use polling instead of native watchers
-  };
+    const compilerOptions = {
+      aggregateTimeout: 300, // wait so long for more changes
+      poll: true, // use polling instead of native watchers
+    };
 
-  // compile server side code
-  serverCompiler.watch(compilerOptions, err => {
-    if (err) {
-      console.log(`Server bundling error: ${JSON.stringify(err)}`);
-      return;
-    }
+    // compile server side code
+    serverCompiler.watch(compilerOptions, err => {
+      if (err) {
+        console.log(`Server bundling error: ${JSON.stringify(err)}`);
+        reject(err);
+        return;
+      }
+
+      clearRequireCache(bundlePath);
+
+      if (!initialLoad) {
+        httpServerInitObject.httpServer.close(() => {
+          httpServerInitObject = initHttpServer(bundlePath);
 
-    clearRequireCache(bundlePath);
+          if (httpServerInitObject) {
+            initialLoad = false;
+            console.log(`Server bundled & restarted ${new Date()}`);
+            resolve(httpServerInitObject.httpServer);
+          } else {
+            // server bundling error has occurred
+            initialLoad = true;
+          }
+        });
 
-    if (!initialLoad) {
-      httpServerInitObject.httpServer.close(() => {
+        // Destroy all open sockets
+        // eslint-disable-next-line no-restricted-syntax
+        for (const socket of httpServerInitObject.sockets.values()) {
+          socket.destroy();
+        }
+      } else {
         httpServerInitObject = initHttpServer(bundlePath);
 
         if (httpServerInitObject) {
           initialLoad = false;
-          console.log(`Server bundled & restarted ${new Date()}`);
+          console.log('Server bundled successfully');
+          resolve(httpServerInitObject.httpServer);
         } else {
           // server bundling error has occurred
           initialLoad = true;
         }
-      });
-
-      // Destroy all open sockets
-      // eslint-disable-next-line no-restricted-syntax
-      for (const socket of httpServerInitObject.sockets.values()) {
-        socket.destroy();
-      }
-    } else {
-      httpServerInitObject = initHttpServer(bundlePath);
-
-      if (httpServerInitObject) {
-        initialLoad = false;
-        console.log('Server bundled successfully');
-      } else {
-        // server bundling error has occurred
-        initialLoad = true;
       }
-    }
+    });
   });
 };
 
 export const watchServerChangesWithDefaultConfig = serverEntryPath => {
```

Comment:
Equally here, please change this to an async function to return the actual httpServer object rather than a promise, thank you.

Reply:
I'm struggling with this. When I try to refactor the code to use async/await rather than promises, I get a "ReferenceError: regeneratorRuntime is not defined". I've tried to fix this by editing the `.babelrc` file [as described here](https://github.com/babel/babel/issues/5085) but this only leads to more errors:

```
TypeError: _webpack.default.HotModuleReplacementPlugin is not a constructor

      22 |   }
      23 | 
    > 24 |   const hmrPlugin = new webpack.HotModuleReplacementPlugin();
         |                     ^
      25 |   if (!plugins) {
      26 |     clientConfig.plugins = [hmrPlugin]; // eslint-disable-line
      27 |   } else {

      at watchClientChanges (src/client/watchClientChanges.js:24:21)
      at Object.test (src/index.test.js:60:7)
```

Would you be okay with me keeping the promises for now and filing a separate issue to convert the code to use async/await?